library/x509_crt.c

```diff
@@ -1893,16 +1940,36 @@ static int x509_crt_check_signature( const mbedtls_x509_crt *child,
                                      mbedtls_x509_crt *parent,
                                      mbedtls_x509_crt_restart_ctx *rs_ctx )
 {
-    const mbedtls_md_info_t *md_info;
     unsigned char hash[MBEDTLS_MD_MAX_SIZE];
-
+    size_t hash_len;
+    /* Note: hash errors can happen only after an internal error */
```

Comment:
This comment is slightly misplaced now; I'd prefer to remove it entirely or move it L.1950 again.

Reply:
Adjusted.